### PR TITLE
Lint all Markdown files

### PR DESCRIPTION
**Pull request by marianhlavac**
Added a linter using **Remark**.

Respecting the [Markdown Style Guide](http://www.cirosantilli.com/markdown-style-guide/), with one exception of list items spacing and indentation.

🗯 **Let's discuss other exceptions, like emphasis markers etc.** We should agree on consistent set of rules before converting the rest of pages.

---

Requires to have Node.js installed. To install the linter, run `npm install` and then to lint use `npm run lint`. *This could be also mentioned in the README.*

### CODE REVIEW

**Comment by hroncok** [comment]
Pffff. I guess if Travis Matrix supports multiple languages, we are good.

I'm not quite confident with the current set of rules. I like to use `*` for lists, `**` for strong and `_` for em.

Also, I don't understand why you say "drop the Yarn", but Travis CI uses it. Care to explain? (I'm not familiar with yarn or remarkable.)

**Comment by syky27** [comment]
OMG, proc travis, kdyz to vsechno ma bejt na Jenkins, tam kde bezej ulohy...

Doesn't Travis has some kind of limitation?

**Comment by hroncok** [comment]
Because this is a public repo and Travis allows us to do this publicly in an easy way without relying on a black box that Jenkins is for most of us.

**Comment by hroncok** [comment]
What kind of limitation?

**Comment by syky27** [comment]
I don't know it was a question...

**Comment by syky27** [comment]
It doesn't really matter to me.

Will the linter work on pull requests?

**Comment by hroncok** [comment]
Why wouldn't it?

**Comment by marianhlavac** [comment]
@hroncok 
> I guess if Travis Matrix supports multiple languages, we are good. 

What other languages we'll be using on CI?

> I'm not quite confident with the current set of rules. I like to use...

`_` is for em is ok, but `*` for lists is not so great choice, I like to keep `*` as reserved symbol for marking strongs and ems, so they can't be confused with each other. But whatever, I don't need to fight for it, if you really like `*` for lists, let's use `*` for lists. Just make some final decision.

> Also, I don't understand why you say "drop the Yarn"

It doesn't matter really, Yarn can be used on projects without Yarn lock files. Yarn makes the installation quicker (paralel downloading etc.) The reason I dropped Yarn is not to force contributors to install another dependency to lint the files locally. But on CI server we can keep using Yarn.

---

@syky27 
Whatever, if we can run linter on Jenkins, it really doesn't matter. Feel free to replace Travis with Jenkins, but @hroncok is right, that Travis is public the same way as this repo is.

**Comment by hroncok** [comment]
> What other languages we'll be using on CI?

Python for Edux integration.

**Comment by jirutka** [comment]
>> I guess if Travis Matrix supports multiple languages, we are good.
>
> What other languages we'll be using on CI?

It doesn’t matter, Travis has basically the same image for all, i.e. even if you declare e.g. python as the project’s language, there will be also nodejs available.

**Comment by hroncok** [comment]
I've created a Travis yaml that now conflicts with this one. @mmajko please try to adapt the file to run the linter.

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
+  - yarn
+  - sudo sed -i 's/trusty/yakkety/' /etc/apt/sources.list
+  - sudo rm -rf /etc/apt/sources.list.d/*
+  - sudo apt-get update -qq
+  - sudo apt-get install -qq pandoc ca-certificates
+  - pip install -r requirements.txt
 script:
-- ":"
+  - yarn lint
+  - ":"
```
Comment (hroncok) [inline .travis.yml:25]: this is a no-op, we don't need to keep it here, it was there just to unset the default value.

File: .travis.yml [hunk 2]
```diff
@@ -1,16 +1,28 @@
...
+addons:
+  apt:
+    sources:
+      - sourceline: deb https://dl.yarnpkg.com/debian/ stable main
+      key_url: https://dl.yarnpkg.com/debian/pubkey.gpg
+    packages:
+      - yarn
 sudo: true
 dist: trusty
+group: deprecated-2017Q3
```
Comment (hroncok) [inline .travis.yml:15]: Why is this needed?

File: .travis.yml [hunk 3]
```diff
@@ -1,16 +1,28 @@
 language: python
 python:
-- '3.6'
+  - '3.6'
+node_js:
```
Comment (hroncok) [inline .travis.yml:4]: I don't believe you can set `node_js` when `language` is se to Python.

File: .travis.yml [hunk 4]
```diff
@@ -21,5 +33,9 @@ deploy:
...
+    - secure: Zhh4osBr4LdLfwC/IggWtP/792gEnJLDHpbayqGm9JfqWPKDVYQK642pO+ofCdvHBehAABghJ5pJIf5Bj9SXpXjgDCPLHBAyW3psURiPgBiBEugMKto7AvNnf/hEjgbxWj+6MN4K9vO/+T1bbkrcOe/kplclmyc9M6a0b9U+ME7MZxMSse1QK0F1You526o5AKllMRfzd04jNnyldW0HiFpMvbKiYNj3krF/zCRzXs+OqaGavS6YLypKosEVaMESdAzdcDixBYXR701GUO0ESVEXhFPIjzBXqQODaH59wfPBhgAFY4PkBihhCnjgWq3tuoMrsZqldVIaQo0Q3ISd+Esn4kYJyQpUO0DaWGouY9Gh77qUeFy+rML1NIQwdPxxTYFInTmnTuiO6OQ6EtbdYpIxXMWtoUsqkVPAQFqAn+8N+3XUXtdaV2oFGObK+adXAgC70rpVzAwuZO6seY4s58sbyzidCxtHVbppTIQLBrGInf+EipOV29VHwcnIfOcn6n3oHYG+9L/m4rhi3TBY3K6+xZEsMllXEmfZRk/tuGTtYgiKXUKN9ABco5qDE8k6o1fUf7k4RBz7hUKE8X8psRIAdBmO6k38ze+BYlzWrFLBTGZVuh0wYnRyb0Z6DVXkWcABUNfXTmpU1L51ypHmTl7+Z1KpItv8T4odl4CRjfU=
+cache:
```
Comment (hroncok) [inline .travis.yml:38]: cache and sudo didn't go along last time I've checked, but it might have changed.

File: .travis.yml [hunk 5]
```diff
@@ -1,16 +1,28 @@
 language: python
 python:
-- '3.6'
+  - '3.6'
```
Comment (hroncok) [inline .travis.yml:3]: Can the file be changed without completely reformatting the YAML?

**Comment by hroncok** [comment]
Ok, I was thinking about using a Markdown linter, and this is what I would like to lint:

 * is the Markdown valid?
 * are there no HTML tags (Edux doesn't like them)?
 * are all link targets defined?
 * are all relative links and image sources existing?
 * are all nested lists always nested by exactly one level?

Those are things I don't mind having linted:

 * line length, end of line marks, utf-8 encoding, no tabs
 * inline URLs

And those are things I probably don't want to lint:

 * what heading style is used (see https://github.com/3DprintFIT/BI-3DT/pull/7#commitcomment-24356742)
 * whether `*` or `-` is used for lists, emphasis, etc.

**Comment by syky27** [comment]
This is in such a huge mess, that i have created new PR #10  where changes are thansparent
